spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,463 @@
+          .defaultValue(19091)
+          .build();
+
+  public static final ConfigOption<Integer> SENTINEL_EXECUTOR_SERVICE_POOL_SIZE =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.health.sentinelExecutorPoolSize")
+          .typeParameterClass(Integer.class)
+          .description(
+              "Size of executor service in Sentinel Managers to check the health "
+                  + "of sentinel resources.")
+          .defaultValue(3)
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<Integer> SENTINEL_RESOURCE_RECONCILIATION_DELAY =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.health.sentinelResourceReconciliationDelaySeconds")
+          .enableDynamicOverride(true)
+          .description(
+              "Allowed max time(seconds) between spec update and reconciliation "
+                  + "for sentinel resources.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(60)
+          .build();
+
+  public static final ConfigOption<Boolean> LEADER_ELECTION_ENABLED =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.leaderElection.enabled")
+          .enableDynamicOverride(false)
+          .description(
+              "Enable leader election for the operator to allow running standby instances. When "
+                  + "this is disabled, only one operator instance is expected to be up and "
+                  + "running at any time (replica = 1) to avoid race condition.")
+          .typeParameterClass(Boolean.class)
+          .defaultValue(false)
+          .build();
+
+  public static final ConfigOption<String> LEADER_ELECTION_LEASE_NAME =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.leaderElection.leaseName")
+          .enableDynamicOverride(false)
+          .description(
+              "Leader election lease name, must be unique for leases in the same namespace.")
+          .typeParameterClass(String.class)
+          .defaultValue("spark-operator-lease")
+          .build();
+
+  public static final ConfigOption<Integer> LEADER_ELECTION_LEASE_DURATION_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.leaderElection.leaseDurationSeconds")
+          .enableDynamicOverride(false)
+          .description("Leader election lease duration in seconds, non-negative.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(1200)
+          .build();
+
+  public static final ConfigOption<Integer> LEADER_ELECTION_RENEW_DEADLINE_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.leaderElection.renewDeadlineSeconds")
+          .enableDynamicOverride(false)
+          .description("Leader election renew deadline in seconds, non-negative.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(600)
```

Comment:
How do we guarantee the relation between `leaseDurationSeconds` and `renewDeadlineSeconds`? 

And, could you describe that we need `leaseDurationSeconds` > `renewDeadlineSeconds` to prevent user mistakes?

Reply:
We do not guarantee actually - in this case RuntimeException would be thrown for invalid config. Added a few more description for renewal deadline.